File: Python/kumarAbhi-kumar/605_Can_Place_Flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        if n == 0:
            return True
        flowr_bed_space = len(flowerbed)
        output = False
        i = 0
        plantable_spaces = 0
        while i < flowr_bed_space:
            if flowerbed[i] == 0:
                if i == 0 or flowerbed[i - 1] == 0:
                    if i == flowr_bed_space - 1 or flowerbed[i + 1] == 0:
                        plantable_spaces += 1
                        if plantable_spaces >= n:
                            output = True
                            break
                        i += 2
                        continue
                i += 1
            else:
                i += 2
        return output

File: CPP/coolakt/605. Can Place Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        for(int i=0; i<flowerbed.size(); i++){
            if(flowerbed[i]==1){
                i++;
            }
            else{
                if((i==flowerbed.size()-1 || flowerbed[i+1] == 0) && (i == 0 || flowerbed[i-1]==0)){
                    flowerbed[i]=1;
                    n--;
                    i++;
                }
            }

        }
        if(n<=0){
            return 1;
        }
        else{
            return 0;
        }
    }
};
